### PR TITLE
SHARE-158 Download several looks

### PR DESCRIPTION
**Pull request by matt-sp**
Implemented a function in the media-library to download multiple looks at once.

Created behat tests.

Modified behat.yml.dist to allow chrome headless to download files which is required for proper testing of this feature.

### CODE REVIEW

**Review by HCrane — APPROVED** [review]
A very nice PR!

File: tests/behat/features/web/mediapackage.feature [hunk 1]
```diff
@@ -56,3 +46,21 @@ Feature:
     And I should see media file with id 7 in category "Luna & Cat Theme Special"
     And I should see 1 media file in category "Luna & Cat Theme Special"
     And I should see 1 media file in category "Bla"
+
+  Scenario: Selecting and deselecting media files for downloading
+    Given I am on "/app/media-library/looks"
```
Comment (dmetzner) [inline tests/behat/features/web/mediapackage.feature:53]: And I wait for the page to be loaded

File: tests/behat/features/web/mediapackage.feature [hunk 2]
```diff
@@ -56,3 +46,21 @@ Feature:
...
+    And I click "#mediafile-1"
+    And I click "#mediafile-5"
+    Then I should see "Download these files"
+    And I click "#mediafile-1"
+    And I click "#mediafile-5"
+    Then I should not see "Download these files"
+
+  Scenario: Downloading mutiple selected files
+    Given I am on "/app/media-library/looks"
```
Comment (dmetzner) [inline tests/behat/features/web/mediapackage.feature:63]: And I wait for the page to be loaded

File: tests/behat/features/web/mediapackage.feature [hunk 3]
```diff
@@ -56,3 +46,21 @@ Feature:
...
+    And I click "#mediafile-1"
+    And I click "#mediafile-5"
+    And I click "#start-downloads"
+    And I wait 5000 milliseconds
```
Comment (dmetzner) [inline tests/behat/features/web/mediapackage.feature]: is it possible to wait for a response or something?

File: tests/behat/features/web/mediapackage.feature [hunk 4]
```diff
@@ -56,3 +46,21 @@ Feature:
     And I should see media file with id 7 in category "Luna & Cat Theme Special"
     And I should see 1 media file in category "Luna & Cat Theme Special"
     And I should see 1 media file in category "Bla"
+
+  Scenario: Selecting and deselecting media files for downloading
+    Given I am on "/app/media-library/looks"
+    And I click "#mediafile-1"
+    And I click "#mediafile-5"
```
Comment (dmetzner) [inline tests/behat/features/web/mediapackage.feature:56]: is something triggered by those clicks for which we may need to wait? Ajax or something else?

**Review by dmetzner — CHANGES_REQUESTED** [review]
Please rebase this PR to current develop and try to use wait conditions in the test if possible

File: tests/behat/features/web/mediapackage.feature (hunk 4 above)
Comment (matt-sp) [reply tests/behat/features/web/mediapackage.feature:56]: The clicks mark certain mediafiles as "selected" which means they get downloaded after clicking the download button. Basically the ids of the files are inserted into a JS array. Thats all.

File: tests/behat/features/web/mediapackage.feature (hunk 3 above)
Comment (matt-sp) [reply tests/behat/features/web/mediapackage.feature]: Now implementend waiting for the downloads to be finished.

File: tests/behat/features/bootstrap/WebFeatureContext.php [hunk 5]
```diff
@@ -1700,49 +1701,34 @@ public function iDownload($arg1)
...
-    Assert::assertEquals('attachment; filename="' . $name . '"', $content_disposition);
-  }
+    $received = false;
+    $file_path = self::BROWSER_DOWNLOAD_DIR . '/' . $name;
 
-  /**
-   * @Given /^the response code should be "([^"]*)"$/
-   * @param $code
-   */
-  public function theResponseCodeShouldBe($code)
```
Comment (dmetzner) [inline tests/behat/features/bootstrap/WebFeatureContext.php]: why have you removed this snippet?
Now the web suite has 1 undefined step:

tests/behat/features/web/project_download.feature line 19.

**Review by dmetzner — CHANGES_REQUESTED** [review]
Please just re-add the now missing code snippet. Everything else works perfectly.

(tests/behat/features/web/project_download.feature:19)

File: tests/behat/features/bootstrap/WebFeatureContext.php (hunk 5 above)
Comment (matt-sp) [reply tests/behat/features/bootstrap/WebFeatureContext.php]: This happened accidentally while rebasing and merging. Already fixed and force pushed.

**Review by dmetzner — APPROVED** [review]

**Pull request closed without merging**